qcs/src/executable.rs

```diff
@@ -81,9 +82,9 @@ impl<'executable> Executable<'executable, '_> {
     /// 1. `quil` is a string slice representing the original program to be run. The returned
     ///     [`Executable`] will only live as long as this reference.
     #[must_use]
-    pub fn from_quil(quil: &'executable str) -> Self {
+    pub fn from_quil<Quil: Into<Arc<str>>>(quil: Quil) -> Self {
```

Comment:
Maybe personal preference, but I tend to shy away from long generic names vs. single letter (`Q` over `Quil`), only because it can be confusing for folks less familiar with the concept & mistake it for a variable or something.

Reply:
I have a pretty strong preference for more descriptive generic names simply because I've found it frustrating to trace types for single letters, particularly when there are multiple generics it becomes pretty unreadable (e.g., in `tokio` their generic type for `response` tends to be `S` because `R` is taken up by `request`... or maybe it's the other way around, I can never remember).

For a short signature like this, it doesn't make much of a difference, but since I want consistency in the naming convention, I try to always stick to naming the generic after the thing it's typing.

Reply:
Ok that's fair, yea not a major thing!